samcli/hook_packages/terraform/hooks/prepare/translate.py

```diff
@@ -403,6 +414,69 @@ def _link_lambda_functions_to_layers(
     ResourceLinker(resource_linking_pair).link_resources()
 
 
+def _link_gateway_methods_to_gateway_rest_apis_call_back(
```

Comment:
Can we rename this function and the first argument to be a little more generic. This callback function is actually usable by any APIGW resource that links on the `RestApiId` field.

Reply:
done